src/sentry/tagstore/snuba/backend.py

```diff
@@ -172,18 +176,34 @@ def __get_tag_keys_for_projects(
             aggregations.append(["uniq", "tags_value", "values_seen"])
         conditions = []
 
-        result = snuba.query(
-            start=start,
-            end=end,
-            groupby=["tags_key"],
-            conditions=conditions,
-            filter_keys=filters,
-            aggregations=aggregations,
-            limit=limit,
-            orderby="-count",
-            referrer="tagstore.__get_tag_keys",
-            **kwargs
-        )
+        filtering_strings = [u"{}={}".format(key, value) for key, value in six.iteritems(filters)]
```

Comment:
This method is used in several contexts. I think we should allow the client to specify a bypass of the cache unless we are extra certain that the cached behavior is ok everywhere.
Also I doubt we would need to cache if the if the user specifies a group_id

Reply:
I'll move the caching up a function to `get_tag_keys_for_projects` so it only impacts a single function.

Reply:
Not sure that is enough, we need to understand who is calling it and if the caching behavior is acceptable in those cases. And if not passing a parameter to skip caching could be fine.